src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/SparkSessionTests.cs

```diff
@@ -65,5 +67,55 @@ public void TestSignaturesV2_4_X()
         {
             Assert.IsType<SparkSession>(SparkSession.Active());
         }
+
+        /// <summary>
+        /// Test CreateDataFrame API.
+        /// </summary>        
+        [Fact]
+        public void TestCreateDataFrame()
+        {                                   
+            var schema = new StructType(new List<StructField>()
+            {
+                new StructField("Name1", new StringType()),
+                new StructField("Name2", new StringType())
+            });
+
+            List<GenericRow> data = new List<GenericRow>();
+            data.Add(new GenericRow(new object[] { "Alice", "Harry" }));
+            data.Add(new GenericRow(new object[] { "Bob", "Mary" }));
+
+            // Calling CreateDataFrame with schema
+            DataFrame df1 = _spark.CreateDataFrame(data, schema);
+            Assert.IsType<DataFrame>(df1);
+            Assert.Equal<StructType>(schema, df1.Schema());
+            var df1Collect = df1.Collect();
+            int numRow = 0;
+            foreach (Row r in df1Collect)
+            {
+                // Checking the values from the dataFrame are same as the values in given data
+                Assert.Equal<object>(r.Values, data[numRow].Values);
+                ++numRow;
+            }
+            Assert.Equal<int>(2, numRow);
+
+            // Calling CreateDataFrame without schema
+            DataFrame df2 = _spark.CreateDataFrame(data);            
+            var expectedSchema = new StructType(new List<StructField>()
+            {
+                new StructField("_1", new StringType()),
+                new StructField("_2", new StringType())
```

Comment:
your implementation also supports IntegerType().  How about modifying `data` to include an IntegerType as well.